### PR TITLE
fix(initrd): Several small fixes

### LINKED ISSUE #904: Kraft run fails if `--rootfs` path to directory has `/` appended
### Describe the bug

Passing a `rootfs` to `kraft run` with a path to a directory works if the path does not end with a `/`, but fails otherwise.

### Steps to reproduce

Clone the [app-nginx](https://github.com/unikraft/app-nginx) repository and build an image for `qemu-x86_64`.
Run `kraft run --rootfs ./rootfs/ -p 8080:80`

The kraftfile:
```
specification: '0.5'
name: nginx
unikraft:
  version: stable
  kconfig:
    - CONFIG_LIBUKLIBPARAM=y
    - CONFIG_LIBUKSCHEDCOOP=y
    - CONFIG_LIBPOSIX_LIBDL=y
    - CONFIG_LIBPOSIX_PROCESS=y
    - CONFIG_LIBPOSIX_USER=y
    - CONFIG_UKSYSINFO=y
    - CONFIG_LIBUKSIGNAL=y
    - CONFIG_LIBUKMMAP=y
    - CONFIG_LIBUKTIME=y
targets:
  - name: nginx-qemu-x86_64-9pfs
    architecture: x86_64
    platform: qemu
    kconfig:
      - CONFIG_PLAT_KVM=y
      - CONFIG_LIBVFSCORE_AUTOMOUNT_ROOTFS=y
      - CONFIG_LIBVFSCORE_ROOTFS_9PFS=y
      - CONFIG_LIBVFSCORE_ROOTFS="9pfs"
      - CONFIG_LIBVFSCORE_ROOTDEV="fs1"
  - name: nginx-qemu-x86_64-initrd
    architecture: x86_64
    platform: qemu
    kconfig:
      - CONFIG_PLAT_KVM=y
      - CONFIG_LIBVFSCORE_AUTOMOUNT_ROOTFS=y
      - CONFIG_LIBVFSCORE_ROOTFS_INITRD=y
      - CONFIG_LIBVFSCORE_ROOTFS="initrd"
  - name: nginx-qemu-arm64-9pfs
    architecture: arm64
    platform: qemu
    kconfig:
      - CONFIG_PLAT_KVM=y
      - CONFIG_ARCH_ARM_64=y
      - CONFIG_MCPU_ARM64_NONE=y
      - CONFIG_LIBVFSCORE_AUTOMOUNT_ROOTFS=y
      - CONFIG_LIBVFSCORE_ROOTFS_9PFS=y
      - CONFIG_LIBVFSCORE_ROOTFS="9pfs"
      - CONFIG_LIBVFSCORE_ROOTDEV="fs1"
  - name: nginx-qemu-arm64-initrd
    architecture: arm64
    platform: qemu
    kconfig:
      - CONFIG_PLAT_KVM=y
      - CONFIG_ARCH_ARM_64=y
      - CONFIG_MCPU_ARM64_NONE=y
      - CONFIG_LIBVFSCORE_AUTOMOUNT_ROOTFS=y
      - CONFIG_LIBVFSCORE_ROOTFS_INITRD=y
      - CONFIG_LIBVFSCORE_ROOTFS="initrd"
  - name: nginx-fc-x86_64-initrd
    architecture: x86_64
    platform: fc
    kconfig:
      - CONFIG_PLAT_KVM=y
      - CONFIG_KVM_BOOT_PROTO_LXBOOT=y
      - CONFIG_KVM_VMM_FIRECRACKER=y
      - CONFIG_LIBVFSCORE_AUTOMOUNT_ROOTFS=y
      - CONFIG_LIBVFSCORE_ROOTFS_INITRD=y
      - CONFIG_LIBVFSCORE_ROOTFS="initrd"
  - name: nginx-fc-arm64-initrd
    architecture: arm64
    platform: fc
    kconfig:
      - CONFIG_PLAT_KVM=y
      - CONFIG_KVM_BOOT_PROTO_LXBOOT=y
      - CONFIG_KVM_VMM_FIRECRACKER=y
      - CONFIG_ARCH_ARM_64=y
      - CONFIG_MCPU_ARM64_NONE=y
      - CONFIG_LIBVFSCORE_AUTOMOUNT_ROOTFS=y
      - CONFIG_LIBVFSCORE_ROOTFS_INITRD=y
      - CONFIG_LIBVFSCORE_ROOTFS="initrd"
libraries:
  musl:
    version: stable
  lwip:
    version: stable
    kconfig:
      - CONFIG_LWIP_UKNETDEV=y
      - CONFIG_LWIP_TCP=y
      - CONFIG_LWIP_WND_SCALE=y
      - CONFIG_LWIP_TCP_KEEPALIVE=y
      - CONFIG_LWIP_THREADS=y
      - CONFIG_LWIP_SOCKET=y
      - CONFIG_LWIP_AUTOIFACE=y
      - CONFIG_LWIP_IPV4=y
      - CONFIG_LWIP_DHCP=y
      - CONFIG_LWIP_DNS=y
  nginx:
    version: stable
    kconfig:
      - CONFIG_LIBNGINX_MAIN_FUNCTION=y
```

### Expected behavior

The image runs normally.

### Which architectures were you using or does this bug affect?

_No response_

### Which operating system were you using or does this bug affect?

_No response_

### Relevant log output

```shell
[    0.000000] ERR:  [libukallocbbuddy] <bbuddy.c @  488> 12000: Failed to add memory region 31a087-31b000: Not enough space after applying page alignments
en1: Added
en1: Interface is up
[    0.104197] ERR:  [libukcpio] <cpio.c @  291> /./rootfs/: Failed to create directory: No such file or directory (2)
[    0.110471] CRIT: [libvfscore] <automount.c @  182> Failed to extract cpio archive to /: -6
[    0.115899] ERR:  [libukboot] <boot.c @  346> Init function at 0x14c730 returned error -1
 E  machine fatally exited
```

### PR DESCRIPTION
**Pull request by craciunoiuc**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

GitHub-Fixes: #904

### CODE REVIEW

**Comment by craciunoiuc** [comment]
```
PS C:\Users\Cezar\kraftkit> go test ./initrd
--- FAIL: TestNewFromDirectory (0.03s)
    directory_test.go:83: Encountered unexpected file in cpio archive: testdata/rootfs
    directory_test.go:94: file [/entrypoint.sh]: got size 27, expected 25
    directory_test.go:94: file [/etc/app.conf]: got size 18, expected 16
    directory_test.go:88: file [/lib/libtest.so.1]: got mode 0100000, expected 0120000
    directory_test.go:91: file [/lib/libtest.so.1]: got linkname "", expected "libtest.so.1.0.0"
    directory_test.go:94: file [/lib/libtest.so.1]: got size 16, expected 0
    directory_test.go:94: file [/lib/libtest.so.1.0.0]: got size 10, expected 9
```

@antoineco The tests still fail on windows but now I think this is because of something else:
- It appends `\CR` when reading files, that's why each line is an extra line
- Links behave differently within Windows

**Comment by antoineco** [comment]
@craciunoiuc good thing that the tests caught that. In the current state of things, an image created on Windows won't run with Unikraft.

File: initrd/directory.go [hunk 1]
```diff
@@ -54,10 +55,11 @@ func (initrd *directory) Build(ctx context.Context) (string, error) {
 	if initrd.opts.output == "" {
 		fi, err := os.CreateTemp("", "")
 		if err != nil {
-			return "", fmt.Errorf("could not make temporary directory: %w", err)
+			return "", fmt.Errorf("could not make temporary file: %w", err)
 		}
 
 		initrd.opts.output = fi.Name()
+		defer fi.Close()
```
Comment (antoineco) [inline initrd/directory.go]: Nice catch ;)

**Review by antoineco — APPROVED** [review]

File: initrd/directory.go (hunk 1 above)
Comment (antoineco) [reply initrd/directory.go]: Wait, this should actually not be in a `defer`. We are re-opening that file immediately below, so we should be closing it right away.

Comment (craciunoiuc) [reply initrd/directory.go]: That's how I had it initially, but wanted to have less lines in the PRs, will change ;)

Comment (craciunoiuc) [reply initrd/directory.go]: Changed

**Review by antoineco — APPROVED** [review]

**Comment by craciunoiuc** [comment]
No fears my son ✝️ push the Merge button

(also add your approved-by in there)

**Pull request merged**